consensus/src/dag/types.rs

```diff
@@ -367,10 +373,16 @@ pub struct CertifiedAck {
     epoch: u64,
 }
 
+impl CertifiedAck {
+    pub fn new(epoch: u64) -> Self {
+        Self { epoch }
```

Comment:
Why do we need the epoch here?

Reply:
every message needs epoch number to decide whether it's behind by epochs